File: 2079.给植物浇水.go
/*
 * @lc app=leetcode.cn id=2079 lang=golang
 *
 * [2079] 给植物浇水
 */

// @lc code=start
func wateringPlants(plants []int, capacity int) int {
	res, i := 0, 0
	c_capacity := capacity
	for i < len(plants) {
		if plants[i] <= capacity {
			res++
			capacity -= plants[i]
			i++
		} else {
			res += 2 * i
			capacity = c_capacity
		}
	}
	return res

}

// @lc code=end

File: 2079.给植物浇水.py
#
# @lc app=leetcode.cn id=2079 lang=python3
#
# [2079] 给植物浇水
#
from typing import List
# @lc code=start
class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        res = 0
        i = 0
        c_capacity = capacity
        while i < len(plants):
            if plants[i] <= capacity:
                res += 1
                capacity -= plants[i]
                i += 1
            else:
                res += 2 * i
                capacity = c_capacity
        return res

s = Solution()
print(s.wateringPlants([2,2,3,3], 5))

# @lc code=end

